work on TSP connectors docs (#196)
skedgo/tripgo-api: docs/extensions.md

3. Enter your `Yelp API key` in your [application credentials](https://tripgo.3scale.net/admin/applications). 
4. Add `allowYelp=true` to your geocoding requests, and then geocoding results will then come augmented with Yelp's API through the TripGo API for your API key (after at most 5 minutes).

---

## TSP connectors

For TSPs that are not yet integrated into our platform, you can add them by:
 

>  - implementing the corresponding API (see the list of API specs available [here](https://skedgo.github.io/TSP-APIs/))
>  - sending to [api@skedgo.com](mailto:api@skedgo.com) the url to access that API

Reviewer: ```suggestion
1. Implementing the corresponding API according to our [API specs](https://skedgo.github.io/TSP-APIs/), and then
2. Sending the URL of where you have deployed your implementation to [api@skedgo.com](mailto:api@skedgo.com).
```

> The process then involves a manual step (which we plan to automate) to add that to our list of TSP connectors and enable it, and that's it, results will then include the new TSP.

Reviewer: @MarianoTucat Can you add some information of what time delay? Does it really not need any way to enable these for an API key, or is that what you mean with "manual step"?

Author: Currently, we need to update the API keys, that may change in the future when we automate this step. BTW, does the bike lane info needs to be precomputed?

 

### How does this work

All TSP connector APIs [share](https://skedgo.github.io/TSP-APIs/shared) in common three endpoints:

- **config**: the list of endpoints that are implemented, similar to [gbfs.json](https://github.com/NABSA/gbfs/blob/v2.0/gbfs.md#gbfsjson)  
- **provider**: the information about the provider, which may include links to mobile apps and deep links (with a pattern); and also the list of modes for the TSP 
- **coverage**: a list of coverage areas of the provider, and for each area, it is possible to define available products and pricing rules    

Config endpoint allows our platform to understand what are the capabilities of the TSP integration of each connector.

> Provider information is what our platform will use to attribute results, and the links that will be included. 

Reviewer: ```suggestion
Provider information is what our platform will use to attribute results. This will be passed on with any API responses that include results from your TSP connector.
```

> Coverage will allow our platform to know in which areas (regions) we can enable the TSP.

Reviewer: This sounds like a manual step ("we can enable"). Is that the case or is it automatic? If it's automatic:

```suggestion
Coverage will allow our platform to know in which areas (i.e., regions) the TSP will be enabled.
```

### Taxi and TNC

> For taxi and TNC providers, we defined a [Taxi](https://skedgo.github.io/TSP-APIs/taxi/) API, and it will allow to:

Reviewer: ```suggestion
For taxi and TNC providers we defined a [Taxi](https://skedgo.github.io/TSP-APIs/taxi/) API which allows you to:
```

> - add static pricing (in `coverage` endpoint), which will be shown as an estimate cost and used by our routing engine when creating trip alternatives.
> - provide realtime ETAs for products
> - provide realtime estimated costs based on specific trip details
> - enable in-app booking capabilities, so users can directly initiate the booking process through our platform or WLs  

Reviewer: ```suggestion
- Add static pricing (in `coverage` endpoint), which will be shown as an estimate cost and used by our routing engine when creating trip alternatives.
- Provide real-time ETAs, which might differ by "product".
- Provide real-time costs estimations based on specific trip details.
- Enable in-app booking capabilities, so users can directly initiate the booking process through our platform.
```

Question: Do we support that in-app booking capabilities already or is that more a coming soon?

> ### Shared bikes and scooters 

Reviewer: ```suggestion
### Shared bikes, scooters and cars
```

> For shared bikes and scooters, we support GBFS standard, with the addition of the shared endpoints mentioned above (except for `config` which is replaced by `gbfs.json`).

Reviewer: ```suggestion
For shared vehicles, such as bikes, scooters or cars, we support the GBFS standard with the addition of the shared endpoints mentioned above (except for `config` which is replaced by the standard `gbfs.json`). This includes support for car-sharing, but be aware that this part of the standard is not finalised yet (as of February 2021).
```

> We also support GBFS sources with only one endpoint (just [free_bike_status.json](https://github.com/NABSA/gbfs/blob/v2.0/gbfs.md#free_bike_statusjson) for example)

Reviewer: ```suggestion
We also support GBFS sources with only one endpoint, for example just the [free_bike_status.json](https://github.com/NABSA/gbfs/blob/v2.0/gbfs.md#free_bike_statusjson) endpoint.
```

     
### Info 

To enhance our platform with more information, we defined an [Info](https://skedgo.github.io/TSP-APIs/info/) API, which allows:
 

>  - reporting real time events on a given location or area. The model follows GTFS-R standard and will be added to our platform responses accordingly.

Reviewer: ```suggestion
 - Reporting real-time events on a given location or area. The model follows GTFS-R standard and will be added to our platform responses accordingly.
```

Also: Can you elaborate which part of the GTFS-R standard or which data model in there? Is it just alerts for example?

>  - integrating bike lanes, which will be used by our cycling results to prioritize those paths.

Reviewer: ```suggestion
 - Integrating bike lanes, which will be used by our cycling results to prioritise those paths.
```
